Updating zip deploy approach with secure blob storage, managed identities, and Elastic Premium

paulyuk opened the pull request:
This pull request modernizes the Azure Function zip deployment solution by shifting from traditional ARM template deployment to a recommended Azure Developer CLI (azd) approach, emphasizing security and best practices. The changes introduce new infrastructure-as-code using Bicep and Azure Verified Modules, leverage managed identities for secure authentication, and update documentation to guide users through the new workflow. Legacy ARM and App Service resources are removed in favor of the new modular, enterprise-grade setup.  Also the PR emphasizes Elastic Premium Linux and Flex Consumption Linux plans which are recommended.

paulyuk committed:
Updated bicep for EP Linux and recompiled JSON files

paulyuk committed:
Larger change to use MI access only and AVM modules

paulyuk committed:
Updating README and refactoring infra

fboucher commented:
@paulyuk I tried to deploy to `eastus` and got an error message about quota. I then tried to deploy to `canadaeast` and got "Requested features are not supported in region."

Could you update the bicep so it shows only the "valide" (at least today's) locations?
Adding instructions about what to do when the Quota error happens would be nice too.

fboucher committed:
Docs: Adds pre-requisites for local development

Adds a 'Pre-requisites' section to the README, guiding users
on setting up their local development environment with the
necessary tools (Azure CLI, Azure Developer CLI, Azure
Functions Core Tools) and Azure login instructions.

fboucher commented:
@paulyuk I used `centralus` and I was able to deploy.
I updated the readme to include the pre- since you mentioned that in your script ;)

However I'm still getting an error. Could it be that something is missing in the bicep files?

```bash
ERROR: error executing step command 'deploy --all': failed deploying service 'api': POST https://func-zipdeployazfuncxxxxxxxxx.scm.azurewebsites.net/api/zipdeploy
--------------------------------------------------------------------------------
RESPONSE 409: 409 Conflict
ERROR CODE UNAVAILABLE
--------------------------------------------------------------------------------
"Run-From-Zip is set to a remote URL using WEBSITE_RUN_FROM_PACKAGE or WEBSITE_USE_ZIP app setting. Deployment is not supported in this configuration."
--------------------------------------------------------------------------------

TraceID: xxxxxxxxx

```

fboucher committed:
Adds instructions for testing deployed function

Adds instructions on how to test the function after deployment to the README.

This allows users to quickly verify the function is running correctly.
